src/reference.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::Arc;
+
+use uuid::Uuid;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct Reference(Arc<String>);
```

Comment:
There are several alternatives to Arc<String> that use only a single heap allocation.
https://crates.io/crates/flexstr

Reply:
Changed to use `flexstr::SharedStr`